components/push/src/subscriber.rs

```diff
@@ -38,10 +38,24 @@ impl PushManager {
     }
 
     // XXX: make these trait methods
-    // XXX: should be called subscribe?
     pub fn subscribe(&mut self, channel_id: &str, scope: &str) -> Result<(RegisterResponse, Key)> {
         let reg_token = self.config.registration_id.clone().unwrap();
         let subscription_key: Key;
+        if let Some(uaid) = self.conn.uaid.clone() {
```

Comment:
We could probably postpone this clone until after get_record returns Some, but it also probably doesn't matter.

Reply:
I was kinda debating that. The first run had just a check to see if `uaid.is_some()`, but then I realized I needed the value, and that to get it for `get_record` I needed to clone uaid to move it into reference, so I just kinda shrugged and went with the easier to read approach.